src/rules/builtins.rs

```diff
@@ -266,19 +258,15 @@ mod tests {
         assert!(result.is_ok(), "{:#?}", result);
 
         let locations = result.unwrap();
-        assert_eq!(locations.len(), 1);
-        assert_eq!(locations[0].range, TextRange::new(16.into(), 19.into()));
-        assert_eq!(&root.syntax().to_string()[locations[0].range], "NVL");
+        check_locations(&locations, r#"[RuleMatch(16..19, "NVL")]"#);
```

Comment:
The `check_locations` seems odd to me. First it probably should have been called `assert_locations` as it's calling the `assert_eq` macro to check the given locations match the expected string.

I don't recommend to stringify the given type `RuleMatch` into its `Debug` representation. This type may seem to work at first, but it's better to test the struct & its fields explicitly. Otherwise it feels like testing a serialized / alternative version instead. A `Debug` implementation is meant to be used as display output for developers (vs `Display` meant to be for users / clients).

As mentioned in the earlier comment, the `check_locations` is probably not necessary, the important part is that the new resulting string looks like expected, no need to check specific text ranges or offsets.